pkg/api/dashboard.go

```diff
@@ -272,6 +276,101 @@ func (hs *HTTPServer) getDashboardHelper(ctx context.Context, orgID int64, id in
 	return queryResult, nil
 }
 
+// swagger:route POST /dashboards/uid/{uid}/trash dashboards restoreDeletedDashboardByUID
+//
+// Restore a dashboard to a given dashboard version using UID.
+//
+// Responses:
+// 200: postDashboardResponse
+// 400: badRequestError
+// 401: unauthorisedError
+// 403: forbiddenError
+// 404: notFoundError
+// 500: internalServerError
+func (hs *HTTPServer) RestoreDeletedDashboard(c *contextmodel.ReqContext) response.Response {
+	uid := web.Params(c.Req)[":uid"]
+	cmd := dashboards.RestoreDeletedDashboardCommand{}
+
+	if err := web.Bind(c.Req, &cmd); err != nil {
+		return response.Error(http.StatusBadRequest, "bad request data", err)
+	}
```

Comment:
Given that `dashboards.RestoreDeletedDashboardCommand` has only one field:
https://github.com/grafana/grafana/blob/cf60048b80af1a107ce2b7dd9950067e25b750fd/pkg/services/dashboards/models.go#L213-L215

we could pass folder UID as an optional query parameter instead and in that case we could change it back to `PATCH` method
WDYT?

Reply:
This is a documentation error, it is a PATCH method. I think using the payload aligns better with RESTful principles. But if you have a better reason to change it by a query param let me know.

Reply:
no, I don't have preference: I only got confused form the above swagger annotation:
`// swagger:route POST /dashboards/uid/{uid}/trash dashboards restoreDeletedDashboardByUID`
since it's `PATCH` should change to:
`// swagger:route PATCH /dashboards/uid/{uid}/trash dashboards restoreDeletedDashboardByUID`